kubernetes/kubernetes.go

```diff
@@ -608,12 +636,27 @@ func ensureMetricServicesAvailable(config KubeAgentConfig) (KubeAgentConfig, err
 	}
 
 	if !config.RetrieveNodeSummaries && !config.CollectHeapsterExport {
+		log.Debugf("Unable to retrieve data due to metrics-agent configuration.\n" +
+			"May be caused by cluster security mis-configurations or the RBAC role in the Cloudability " +
+			"namespace needs to be updated.\n" +
+			"Please confirm with your cluster security administrators that the " +
+			"RBAC role is able to work within your cluster's security configurations.")
 		return config, fmt.Errorf("unable to retrieve node summaries or heapster export: %s", err)
 	}
 
 	return config, nil
 }
 
+func handleNodeSourceError(err error) string {
+	var error string
+	if strings.Contains(err.Error(), "Please verify RBAC roles") {
+		error = fmt.Sprintf(rbacError, kbProvisionURL)
+	}
+	errStr := "Warning non-fatal error: Agent error occurred verifying node source metrics: %v\n" +
+		"For more information see: %s"
+	return error + fmt.Sprintf(errStr, err, kbErrorURL)
```

Comment:
⭐ 
Ah, nice. I like the way you coded this so if the error is not related to RBAC roles, then the final error will be prefixed by an empty string - clever! 😄 One suggestion for improvement: It's a bit confusing to read though with the `error` var naming as-is. What do you think about changing that var to be something more descriptive, like: `optionalRBACErrMsg` or `optionalProvisioningErrMsg`

Reply:
Ah right, that's a reserved word, I'll fix it.